addons/account/views/account_move_views.xml

```diff
@@ -1178,6 +1178,7 @@
                     <field name="partner_id" operator="child_of"/>
                     <field name="invoice_user_id" string="Salesperson" domain="[('share', '=', False)]"/>
                     <field name="date" string="Period"/>
+                    <field name="line_ids" string="Product" filter_domain="[('line_ids.product_id', 'ilike', self)]"/>
```

Comment:
You don't have the autocompletion on the product like that, so might as well keep it simple and more generic...
```suggestion
                    <field name="line_ids" string="Invoice Line"/>
```